pkg/cluster/px/px.go

```diff
@@ -524,21 +527,35 @@ func (ops *pxClusterOps) upgradePX(newVersion string) error {
 			return nil, true, errNoPXDaemonset
 		}
 
+		pxImage := fmt.Sprintf("%s:%s", spec.PXImage, spec.PXTag)
+		ociImage := fmt.Sprintf("%s:%s", spec.OCIMonImage, spec.OCIMonTag)
+
 		for _, ds := range dss {
 			skip := false
 			logrus.Infof("Upgrading PX daemonset: [%s] %s to version: %s", ds.Namespace, ds.Name, newVersion)
 			dsCopy := ds.DeepCopy()
 			for i := 0; i < len(dsCopy.Spec.Template.Spec.Containers); i++ {
 				c := &dsCopy.Spec.Template.Spec.Containers[i]
 				if c.Name == pxDaemonsetContainerName {
-					if c.Image == newVersion {
+					if c.Image == ociImage {
 						logrus.Infof("Skipping upgrade of PX daemonset: [%s] %s as it is already at %s version.",
 							ds.Namespace, ds.Name, newVersion)
 						expectedGenerations[ds.UID] = ds.Status.ObservedGeneration
 						skip = true
 					} else {
 						expectedGenerations[ds.UID] = ds.Status.ObservedGeneration + 1
-						c.Image = newVersion
+						c.Image = ociImage
+						if spec.OCIMonTag != spec.PXTag {
+							c.Env = setEnv(c.Env, "PX_IMAGE", pxImage)
+						}
+						if len(ops.dockerRegistrySecret) > 0 {
+							secret, err := ops.k8sOps.GetSecret(ops.dockerRegistrySecret, ds.Namespace)
+							if err != nil {
+
+							}
+							c.Env = setEnv(c.Env, "REGISTRY_USER", string(secret.Data["REGISTRY_USER"]))
```

Comment:
Remove `REGISTRY_USER` and `REGISTRY_PASS` logic. This should be handled by the installer ( spawn in case of automation). The upgrade script should be concerned only with image versions rather than adding add.